features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+                    uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+                    method: 'PUT',
+                    auth: {
+                        bearer: this.jwt
+                    },
+                    body: {
+                        pipelineIds: [pipelineId]
+                    },
+                    json: true
+                }).then((response) => {
+                    Assert.strictEqual(response.statusCode, 200);
+                });
+            });
+    });
+
+    Given(/^they have a collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.firstCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    Given(/^they have a collection "anotherCollection"$/, function step() {
+        return createCollection.call(this, { name: 'anotherCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.secondCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.body.length, 2);
+            Assert.strictEqual(response.body[0].name, 'anotherCollection');
+            Assert.strictEqual(response.body[1].name, 'myCollection');
+        });
+    });
+
+    When(/^they delete that collection$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'DELETE',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.oneOf(response.statusCode, [404, 204]);
```

Comment:
The collection should actually exist. `404` should not be an acceptable status code.